app/exec/build-tasks-upload.ts

```diff
@@ -25,33 +27,55 @@ export class BuildTaskUpload extends cmdm.TfCommand {
     public exec(args: string[], options: cm.IOptions): any {
         var taskPath = args[0] || options['taskPath'];
         this.checkRequiredParameter(taskPath, 'taskPath', 'path to the task folder');
+        var deferred = Q.defer<agentifm.TaskDefinition>();
+        fs.readFile(path.join(taskPath, 'task.json'), 'utf8', (err, data) => {
+            if(err) {
+                return console.log(err);
```

Comment:
no - this needs to call reject(err).  let the cli infra handle messaging the errors